### DISCUSSION TITLE
How can I compile libsodium statically using aarch64-elf

### DISCUSSION THREAD

**Discussion by MasterAwesome** [jedisct1/libsodium#1328] [Q&A]
I'm trying to compile for a target that has a compiler that can produce only static libraries, although automake seems to not like this since it tries to link a basic C program and I cannot run `./configure --host=aarch64-elf --disable-shared --enable-static <CC and other compiler variables>` 

The goal I'm trying to achieve is to have libsodium be a static library, my target does support libc (although unsure if libsodium depends on it).

**Comment by jedisct1** [comment]
What happens when the `./configure` command is run? 

Even if it doesn't complete, it creates a `config.log` file that is usually very useful to understand what it tried to do and what failed.

**Comment by jedisct1** [comment]
Can you compile a minimal application?

Put this into `test.c`:

```c
int main(void)
{
    return 0;
}
```

And try to compile with:

```sh
aarch64-elf-gcc test.c
```

Does that work?

**Comment by jedisct1** [comment]
Worst case scenario, you can do something like:

```sh
find . -name '*.o' -exec rm -f {} \;

aarch64-elf-gcc $(find src/libsodium -type f -name '*.c') \
    -I src/libsodium/include -I src/libsodium/include/sodium \
    -I src/libsodium/include/sodium/private -Os -c

aarch64-elf-ar rcs libsodium.a $(find . -type f -name '*.o')
```

But many symbols from libc may be missing.

**Reply by mscoder1987** [reply]
Please allow me to ask a similar question in this thread. I used 'objdump -t libsodium.a' to check the symbol table of the Android version static library, and I was able to find the 'sodium_init' method. However, when I link it to my dynamic 'so' library, I always get the 'ld: error: undefined symbol: sodium_init' error.
